Tests/WorkspaceTests/WorkspaceTests.swift

```diff
@@ -559,6 +559,81 @@ final class WorkspaceTests: XCTestCase {
             }
         }
     }
+    
+    func testCanUneditRemovedDependencies() {
+
+        mktmpdir { path in
+            let manifestGraph = try MockManifestGraph(at: path,
+                rootDeps: [
+                    MockDependency("A", version: Version(1, 0, 0)..<Version(1, .max, .max)),
+                ],
+                packages: [
+                    MockPackage("A", version: v1)
+                ]
+            )
+            
+            let delegate = TestWorkspaceDelegate()
+            let repoPath = AbsolutePath(manifestGraph.repo("A").url)
+            
+            func createWorkspace(with manifestGraph: MockManifestGraph) throws -> Workspace {
+                return  Workspace.createWith(rootPackage: path, manifestLoader: manifestGraph.manifestLoader, delegate: delegate)
+            }
+            
+            do {
+                var workspace = try createWorkspace(with: manifestGraph)
+                
+                // Load the package graph.
+                let diagnostics = DiagnosticsEngine()
+                workspace.loadPackageGraph(rootPackages: [path], diagnostics: diagnostics)
+            
+                let rootManifests = workspace.loadRootManifests(packages: [path], diagnostics: diagnostics)
+                let manifests = workspace.loadDependencyManifests(rootManifests: rootManifests, diagnostics: diagnostics)
+                
+                guard let aManifest = manifests.lookup(manifest: "A") else {
+                    return XCTFail("Expected manifest for package A not found")
+                }
+                
+                func getDependency(_ manifest: Manifest) -> ManagedDependency {
+                    return workspace.managedDependencies[manifest.url]!
+                }
+                
+                let dependency = getDependency(aManifest)
+        
+                // Put the dependency in edit mode at its current revision.
+                workspace.edit(
+                    packageName: "A",
+                    revision: dependency.checkoutState!.revision,
+                    diagnostics: diagnostics
+                )
+                let editedDependency = getDependency(aManifest)
+                let editRepoPath = workspace.editablesPath.appending(editedDependency.subpath)
+
+                // Its basedon should not be nil
+                XCTAssertNotNil(editedDependency.basedOn)
+                
+                // Change the manifest to not require the dependency.
```

Comment:
Technically this should say `// Create an empty root to remove the dependency requirement`